services/tests/test_ta_timeseries/test_ta_timeseries.py

```diff
@@ -0,0 +1,387 @@
+                    "filename": "test_filename",
+                    "build_url": None,
+                },
+            ],
+        },
+        flaky_test_ids={
+            calc_test_id("flaky_test_name", "test_classname", "test_suite")
+        },
+    )
+
+    testruns = get_testruns_for_flake_detection(
+        1,
+        {
+            calc_test_id("flaky_test_name", "test_classname", "test_suite"),
+        },
+    )
+    assert len(testruns) == 3
+    assert testruns[0].outcome == "failure"
+    assert testruns[0].failure_message == "failure_message"
+    assert testruns[0].name == "test_name"
+    assert testruns[1].outcome == "flaky_failure"
+    assert testruns[1].failure_message == "failure_message"
+    assert testruns[1].name == "flaky_test_name"
+    assert testruns[2].outcome == "pass"
+    assert testruns[2].failure_message == "failure_message"
+    assert testruns[2].name == "flaky_test_name"
+
+
+@pytest.mark.django_db(databases=["timeseries"])
+@travel(datetime(2025, 1, 1), tick=False)
+def test_update_testrun_to_flaky():
+    insert_testrun(
+        timestamp=datetime.now(),
+        repo_id=1,
+        commit_sha="commit_sha",
+        branch="branch",
+        upload_id=1,
+        flags=["flag1", "flag2"],
+        parsing_info={
+            "framework": "Pytest",
+            "testruns": [
+                {
+                    "name": "test_name",
+                    "classname": "test_classname",
+                    "computed_name": "computed_name",
+                    "duration": 1.0,
+                    "outcome": "failure",
+                    "testsuite": "test_suite",
+                    "failure_message": "failure_message",
+                    "filename": "test_filename",
+                    "build_url": None,
+                },
+            ],
+        },
+    )
+    update_testrun_to_flaky(
+        datetime.now(),
+        calc_test_id("test_name", "test_classname", "test_suite"),
+        calc_flags_hash(["flag1", "flag2"]),
+    )
+    testrun = Testrun.objects.get(
+        name="test_name",
+        classname="test_classname",
+        testsuite="test_suite",
+    )
+    assert testrun.outcome == "flaky_failure"
+
+
+@pytest.mark.integration
+@pytest.mark.django_db(databases=["timeseries"], transaction=True)
+def test_get_testrun_summary():
+    connection = connections["timeseries"]
+    with connection.cursor() as cursor:
+        cursor.execute(
+            """
+            TRUNCATE TABLE timeseries_testrun;
+            TRUNCATE TABLE timeseries_testrun_summary_1day;
+            """
+        )
+        cursor.execute(
+            """
+            SELECT remove_continuous_aggregate_policy('timeseries_testrun_summary_1day');
+            select add_continuous_aggregate_policy(
+                'timeseries_testrun_summary_1day',
+                start_offset => '7 days',
+                end_offset => NULL,
+                schedule_interval => INTERVAL '1 milliseconds'
```

Comment:
this is a bit weird. given that the default is set to one day, does that mean these aggregates are only updated once a day, and if you query these during the day, they would not give you any result for "today"?

I think the "continuous aggregation" should indeed be continuous and live.
meaning that when you `INSERT` into the testruns table, the materialized view with the aggregate should yield aggregates for that **immediately**. without having to mess with some kind of setting, and without having to put a `sleep` into the test.